google/resource_dns_record_set.go

```diff
@@ -55,6 +57,147 @@ func resourceDnsRecordSet() *schema.Resource {
+	// U+3002 (ideographic full stop), U+FF0E (fullwidth full
+	// stop), U+FF61 (halfwidth ideographic full stop).
+	re, err := regexp.Compile("[\u002e\u3002\uFF0E\uFF61]$")
+	if err != nil {
+		errors = append(errors, fmt.Errorf("Internal error compiling regexp: %s",
+			err))
+		re = nil
+	}
+	if re != nil && !re.MatchString(name) {
+		errors = append(errors,
+			fmt.Errorf("DNS name must be fully qualified with a trailing dot"))
+	}
+
+	dot_re, err := regexp.Compile("[\u002e\u3002\uFF0E\uFF61]")
+	if err != nil {
+		errors = append(errors, fmt.Errorf("Internal error compiling regexp: %s",
+			err))
+		dot_re = nil
+	}
+
+	verboten_re, err := regexp.Compile("[\x00-\x2C\x2E-\x2F\x3A-\x40\x5B-\x60\x7B-\x7F]+")
+	if err != nil {
+		errors = append(errors, fmt.Errorf("Internal error compiling regexp: %s",
+			err))
+		verboten_re = nil
+	}
+
+	all_digit_re, err := regexp.Compile("^[[:digit:]]+$")
+	if err != nil {
+		errors = append(errors, fmt.Errorf("Internal error compiling regexp: %s",
+			err))
+		all_digit_re = nil
+	}
+
+	space_re, err := regexp.Compile("[[:space:]]+")
+	if err != nil {
+		errors = append(errors, fmt.Errorf("Internal error compiling regexp: %s",
+			err))
+		space_re = nil
+	}
+
+	underscore_re, err := regexp.Compile("_+")
+	if err != nil {
+		errors = append(errors, fmt.Errorf("Internal error compiling regexp: %s",
+			err))
+		underscore_re = nil
+	}
+
+	var labels []string
+	if dot_re != nil {
+		labels := dot_re.Split(name, -1)
+		if len(labels) < 2 {
+			errors = append(errors,
+				fmt.Errorf("DNS name must not be a top level domain."))
+			labels = nil
+		}
+	}
+
+	for index, label := range labels {
+		// The final label is empty
+		if index+1 == len(labels) && len(label) < 1 {
+			errors = append(errors, nil)
+		}
+
+		if len(label) < 1 {
+			errors = append(errors, fmt.Errorf("subdomains must not be empty"))
+		}
+
+		s := []rune(label)
+		if len(s) > 63 {
+			errors = append(errors,
+				fmt.Errorf("subdomains must not have more than 63 characters: %d.",
+					len(s)))
+		}
+		if string(s[0]) == "-" {
+			errors = append(errors, fmt.Errorf("subdomains must not start with a hyphen."))
+		}
+
+		if string(s[2:3]) == "--" {
+			errors = append(errors, fmt.Errorf("subdomains must not mask encoded IDNA: %s.",
+				label))
+		}
+
+		if string(s[len(s)-1]) == "-" {
+			errors = append(errors, fmt.Errorf("subdomains must not end with a hyphen."))
+		}
+
+		if space_re != nil && space_re.MatchString(label) {
+			errors = append(errors,
+				fmt.Errorf("Subdomain labels cannot contain whitespace: %s",
+					label))
+		}
+
+		if all_digit_re != nil && all_digit_re.MatchString(label) {
+			errors = append(errors,
+				fmt.Errorf("Subdomain labels cannot all be digits: %s",
+					label))
+		}
+
+		if verboten_re != nil {
+			bad_chars := verboten_re.FindAllStringSubmatch(label, -1)
+			if bad_chars != nil {
+				errors = append(errors,
+					fmt.Errorf("Illegal ASCII character(s) % q  in domain name.",
```

Comment:
looks like you accidentally got a space between the % and the q

Reply:
fixed